crates/client/src/client/async_client.rs

```diff
@@ -851,20 +850,11 @@ where
             return Poll::Ready(None);
         }
 
-        let message = if let Some(response) = ready!(self.state.inner().poll_next_unpin(cx)) {
-            Some(match response {
-                Ok(ok) => {
-                    if let Err(e) = self.state.process(ok.answers()) {
-                        Err(e)
-                    } else {
-                        Ok(ok)
-                    }
-                }
-                Err(e) => Err(e.into()),
-            })
-        } else {
-            None
-        };
+        let message = ready!(self.state.inner().poll_next_unpin(cx)).map(|response| {
+            let ok = response?;
+            self.state.process(ok.answers())?;
+            Ok(ok)
+        });
```

Comment:
@djc finally found a compact and clear way to do it. When `try { }` will land the awkward` `map` would retire. ;)

May need a carefull review.

Reply:
if needed `ready!` result could be in a separated `let` i am quite a fan of binding everything it help a lot debugging experience.

Reply:
This looks like a good cleanup.